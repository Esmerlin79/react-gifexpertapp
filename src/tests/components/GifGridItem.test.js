import GifGridItem from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Test <GifGridItem />', () => {

    const title = 'a title';
    const url = 'https://localhost/test.jpg';

    let wrapper = shallow(
        <GifGridItem title={ title } url={ url } />
    )
    beforeEach(() => {
        wrapper = shallow(
            <GifGridItem  title={ title } url={ url }/>
        )
    })

    test('should display the component correctly ', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('should have a paragraph with the title ', () => {
        const paragraph = wrapper.find('p');
        expect( paragraph.text().trim() ).toBe( title );
    })
    
    test('should have the image and alt whit the url and title ', () => {
        const img = wrapper.find('img');
        // console.log( img.props() );
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })
    test('should have the class animate__fadeIn  ', () => {
        const div = wrapper.find('div');
        // expect( div.hasClass('animate__fadeIn') ).toBe( true )
        const className = div.prop('className');
        expect( className.includes('animate__fadeIn') ).toBe( true )
    })
    
})
