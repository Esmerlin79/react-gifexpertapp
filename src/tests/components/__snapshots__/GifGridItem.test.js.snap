// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> should display the component correctly  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="a title"
    src="https://localhost/test.jpg"
  />
  <p>
     
    a title
  </p>
</div>
`;
